pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1626,18 +1718,18 @@ func getGceRegion(zone string) (string, error) {
 }
 
 // Converts a Disk resource to an AttachedDisk resource.
-func (gce *GCECloud) convertDiskToAttachedDisk(disk *compute.Disk, readWrite string) *compute.AttachedDisk {
+func (gce *GCECloud) convertDiskToAttachedDisk(disk *gceDisk, readWrite string) *compute.AttachedDisk {
 	return &compute.AttachedDisk{
 		DeviceName: disk.Name,
 		Kind:       disk.Kind,
 		Mode:       readWrite,
-		Source:     "https://" + path.Join("www.googleapis.com/compute/v1/projects/", gce.projectID, "zones", gce.zone, "disks", disk.Name),
+		Source:     "https://" + path.Join("www.googleapis.com/compute/v1/projects/", gce.projectID, "zones", disk.Zone, "disks", disk.Name),
 		Type:       "PERSISTENT",
 	}
 }
 
-func (gce *GCECloud) ListClusters() ([]string, error) {
-	list, err := gce.containerService.Projects.Zones.Clusters.List(gce.projectID, gce.zone).Do()
+func (gce *GCECloud) listClustersInZone(zone string) ([]string, error) {
+	list, err := gce.containerService.Projects.Zones.Clusters.List(gce.projectID, zone).Do()
```

Comment:
@brendandburns should have a good answer since he added it in https://github.com/kubernetes/kubernetes/pull/2355.